hack/run-e2e-kind.sh

```diff
@@ -37,6 +37,7 @@ function get-k8s-server-version {
 
 function install-volcano {
   install-helm
+  install-ginkgo
```

Comment:
There is `ginkgo` installation operation in the script running e2e, is it not effective?

https://github.com/volcano-sh/volcano/blob/082d372f88c436f306766ede0d4414d8fb4613eb/hack/run-e2e-kind.sh#L124

Reply:
Actually it doesn't work and warns that `package golang.org/x/tools/go/ast/inspector: /go/src/golang.org/x/tools exists but /go/src/golang.org/x/tools/.git does not - stale checkout?`

Maybe using install command of `go install github.com/onsi/ginkgo/v2/ginkgo` is the right way?

docs from https://go.dev/doc/go-get-install-deprecation
```
Starting in Go 1.17, installing executables with go get is deprecated. go install may be used instead.

In Go 1.18, go get will no longer build packages; it will only be used to add, update, or remove dependencies in go.mod. Specifically, go get will always act as if the -d flag were enabled.
```

Reply:
What is the go version of your current environment? The go version in the e2e basic environment is 1.19.x. You can use this command to automatically install ginkgo.

Also, you can try `go install` instead to see if it works

Reply:
I use go1.20 and go install works, so I submit this pr. Maybe `go install` is more compareable than `go get`. 
Another way is to replace `go get` with `install-ginkgo `function

Reply:
![image](https://github.com/volcano-sh/volcano/assets/21003791/87f8a94d-19a4-47a5-aab5-cae3ce014de7)

If you want to validate all the preocess, please find a clean enviroment and start at first use.

Reply:
go install in thie pr will also failed on macos because GOBIN is set in makefile, as issue #2894 describes.